src/html5/parser/document.rs

```diff
@@ -418,6 +438,21 @@ impl Document {
     pub fn has_cyclic_reference(&self, node_id: NodeId, parent_id: NodeId) -> bool {
         has_child_recursive(&self.arena, node_id, parent_id)
     }
+
+    /// Check if a given node's children contain a certain tag name
+    pub fn contains_child_tag(&self, node_id: NodeId, tag: &str) -> bool {
+        if let Some(node) = self.get_node_by_id(node_id) {
+            for child_id in &node.children {
+                if let Some(child) = self.get_node_by_id(*child_id) {
+                    if child.name == tag {
+                        return true;
+                    }
```

Comment:
Replace this with `return child.name == tag`

Reply:
Nvm, this change needs to be reverted. It cases an early return when finding the first child not equal to the search tag... guess my brain wasn't working.